Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -149,9 +190,84 @@ export class ArtifactEngine {
         }
     }
 
-    private artifactItemStore: ArtifactItemStore = new ArtifactItemStore();
+    createNewHashMap(sourceProvider: models.IArtifactProvider, itemsToProcess: models.ArtifactItem[], artifactEngineOptions?: ArtifactEngineOptions): Promise<string> {
+        return new Promise((resolve,reject) => {
+            if(!artifactEngineOptions.enableIncrementalDownload) {
+                this.newHashMap = {};
+                resolve();
+            }
+            else {
+                sourceProvider.getArtifactItems(itemsToProcess[0]).then((items: models.ArtifactItem[]) => {
+                    sourceProvider.getArtifactItem(items.find(x => path.normalize(x.path) === path.join(itemsToProcess[0].path, 'artifact-metadata.csv'))).then((hashStream : NodeJS.ReadableStream) => {                        
+                        var newHashPromise = new Promise((resolve) => {
+                            var newHash = readline.createInterface ({
+                                input : hashStream
+                            });
+                    
+                            newHash.on('line', (line) => {
+                                var words = line.split(',');
+                                this.newHashMap[words[0]] = words[1];
+                            });
+                
+                            newHash.on('close', () => {
+                                resolve();
+                            });
+                        });
+                        newHashPromise.then(() => {
+                            this.artifactItemStore.setHashMap(this.newHashMap)
+                            resolve();                                   
+                        });                            
+                    }, (err) => {
+                        Logger.logInfo("Incremental Download Failed. Downloading normally.")
+                        artifactEngineOptions.enableIncrementalDownload = false;
+                        this.newHashMap = {};
+                        resolve();
+                    });
+                }, (err) => {
+                    Logger.logInfo("Incremental Download Failed. Downloading normally.")
+                    artifactEngineOptions.enableIncrementalDownload = false;
+                    this.newHashMap = {};
+                    resolve();
+                });
+            }
+        });
+    }
+
+    cacheUpdation(sourceProvider: models.IArtifactProvider, destProvider: models.IArtifactProvider, resolve) {    
+        if(this.artifactItemStore._downloadTickets.find(x => x.state === models.TicketState.Failed)) {
```

Comment:
cacheUpdation shouldn't have logic to figure out if the download failed, instead we should just call the function only on success.